components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/ContextMenuFeature.kt

```diff
@@ -119,11 +119,16 @@ class ContextMenuFeature(
  * needs to be shown.
  */
 internal class ContextMenuObserver(
-    sessionManager: SessionManager,
+    private val sessionManager: SessionManager,
     private val feature: ContextMenuFeature
 ) : SelectionAwareSessionObserver(sessionManager) {
     override fun onLongPress(session: Session, hitResult: HitResult): Boolean {
         feature.onLongPress(session, hitResult)
         return false
     }
+
+    fun start(sessionId: String?) {
+        val session = sessionId?.let { sessionManager.findSessionById(sessionId) }
+        session?.let { observeFixed(it) } ?: observeSelected()
```

Comment:
nit: I guess this could have been just an if-else too, or not fancy enough? :D

Reply:
I was actually going to file an issue that we do this check for the `observeFixed`/`observeSelected` quite often for our features that we should abstract out off needing to write these lines over and over again.

I've also grown to dislike typing 'null' so I try to get away from it and let the compiler write that in the bytecode instead. 😅

Reply:
Heh. Optimize for reading not writing 😜